set-up.R

```diff
@@ -16,25 +17,133 @@ pkgs <- c(
   "stplanr", # Sustainable transport planning with R
   "tidyr", # tidies up your data!
   "readr", # reads your data fast
-  "knitr", # for knitting it all together
-  "geojsonio",
-  "rmapshaper" # To simplify rnet
-  )
-# Which packages do we require?
-# lapply(pkgs, library, character.only = T)
+  "knitr" # for knitting it all together
+)
+
+# Github packages to install.
+ghpkgs <- c(
+  "ateucher/rmapshaper"
+)
+
+# Package names for GitHub
+ghpkgns <- c(
+  "rmapshaper"
+)
+
+# Ensure packages are up-to-date.
+cat("It is strongly recommended that you update your installed packages.\n");
+update.packages()
+
+##
+## Problematic packages
+##
+## Package in this section may not install due to missing libraries or toolchains, since this
+## is usually catastrophic, we install them separately and issue instructions for fixing the
+## problem here.
+##
+
+# Devtools may fail to install due to missing libraries or toolchains; this is catastrophic,
+# so check for that here and abort setup and execution if this is the case.
+if(!require(devtools)) {
+  install.packages("devtools", clean = TRUE)
+  if (!require(devtools)) {
+    cat("---- COULD NOT INSTALL PCT DEPENDENCY: rgdal ----\n")
+    cat("This is usually caused by missing libcurl headers.  Try installing:\n")
+    cat("\t* deb: libcurl4-openssl-dev (Debian, Ubuntu, etc)\n")
+    cat("\t* rpm: libcurl-devel (Fedora, CentOS, RHEL)\n")
```

Comment:
I love the fact that this is giving clear help to the end user but this is quite allot for us to maintain.  If `devtools` updates to use a system dependency (such as `lib-new-library`) then this message becomes confusing.  The https://github.com/jeroenooms/curl package is installed by `devtools` and it prints this warning.  I think it is useful to stop and direct the user to look at the warnings but I don't think we should repeat them: they might change and we would always have to keep ours upto date.